Sources/FoundationEssentials/FileManager/FileManager+SymbolicLinks.swift

```diff
@@ -97,6 +121,79 @@ extension _FileManagerImpl {
     }
     
     func destinationOfSymbolicLink(atPath path: String) throws -> String {
+#if os(Windows)
+        return try path.withNTPathRepresentation {
+            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+            guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            guard faAttributes.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT == FILE_ATTRIBUTE_REPARSE_POINT else {
+                throw CocoaError.errorWithFilePath(path, win32: ERROR_BAD_ARGUMENTS, reading: true)
+            }
+
+            let hFile: HANDLE = CreateFileW($0, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nil, OPEN_EXISTING, FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS, nil)
+            if hFile == INVALID_HANDLE_VALUE {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+            defer { CloseHandle(hFile) }
+
+            let buffer: UnsafeMutableRawBufferPointer =
```

Comment:
It looks like this buffer is never deallocated. Would it be better to use `withUnsafeTemporaryAllocation` here so that it's stack allocated if possible? If not, we should add a deferred call to `deallocate` to make sure the memory is free'd

Reply:
Yeah, I think that `withUnsafeTemporaryAllocation` makes sense.